diff --git a/node_modules/@inertiajs/core/dist/index.esm.js b/node_modules/@inertiajs/core/dist/index.esm.js
index 3fa968f..bba3c2f 100644
--- a/node_modules/@inertiajs/core/dist/index.esm.js
+++ b/node_modules/@inertiajs/core/dist/index.esm.js
@@ -512,7 +512,6 @@ var CurrentPage = class {
       const isServer2 = typeof window === "undefined";
       const location = !isServer2 ? window.location : new URL(page2.url);
       const scrollRegions = !isServer2 && preserveScroll ? history.getScrollRegions() : [];
-      replace = replace || isSameUrlWithoutHash(hrefToUrl(page2.url), location);
       return new Promise((resolve) => {
         replace ? history.replaceState(page2, () => resolve(null)) : history.pushState(page2, () => resolve(null));
       }).then(() => {
@@ -881,6 +880,14 @@ var EventHandler = class {
     return () => document.removeEventListener(type, listener);
   }
   handlePopstateEvent(event) {
+    if (page.__jodi_isHistoryModal) {
+        delete page.__jodi_isHistoryModal;
+        const url = hrefToUrl(page.get().url)
+        url.hash = ""
+        const page2 = { ...page.get(), url: url.pathname + url.search }
+        page.setQuietly(page2, { preserveState: true });
+        return
+    }
     const state = event.state || null;
     if (state === null) {
       const url = hrefToUrl(page.get().url);
@@ -1360,6 +1367,7 @@ var RequestParams = class _RequestParams {
         onCancel: this.wrapCallback(params, "onCancel"),
         onSuccess: this.wrapCallback(params, "onSuccess"),
         onError: this.wrapCallback(params, "onError"),
+        onInvalid: this.wrapCallback(params, "onInvalid"),
         onCancelToken: this.wrapCallback(params, "onCancelToken"),
         onPrefetched: this.wrapCallback(params, "onPrefetched"),
         onPrefetching: this.wrapCallback(params, "onPrefetching")
@@ -1654,9 +1662,9 @@ var Response = class _Response {
       ...this.response,
       data: this.getDataFromResponse(this.response.data)
     };
-    if (fireInvalidEvent(response)) {
+    if (this.requestParams.all().onInvalid(response) !== false && fireInvalidEvent(response)) {
       return config.get("future.useDialogForErrorModal") ? dialog_default.show(response.data) : modal_default.show(response.data);
-    }
+    };
   }
   isInertiaResponse() {
     return this.hasHeader("x-inertia");
@@ -2095,13 +2103,13 @@ var Router = class {
       keepAlive: options.keepAlive ?? false
     });
   }
-  visit(href, options = {}) {
+  async visit(href, options = {}) {
     const visit = this.getPendingVisit(href, {
       ...options,
       showProgress: options.showProgress ?? !options.async
     });
     const events = this.getVisitEvents(options);
-    if (events.onBefore(visit) === false || !fireBeforeEvent(visit)) {
+    if (await events.onBefore(visit) === false || !fireBeforeEvent(visit)) {
       return;
     }
     const requestStream = visit.async ? this.asyncRequestStream : this.syncRequestStream;
@@ -2200,7 +2208,7 @@ var Router = class {
     return page.resolve(component);
   }
   replace(params) {
-    this.clientVisit(params, { replace: true });
+    return this.clientVisit(params, { replace: true });
   }
   replaceProp(name, value, options) {
     this.replace({
@@ -2240,7 +2248,7 @@ var Router = class {
     );
   }
   push(params) {
-    this.clientVisit(params);
+    return this.clientVisit(params);
   }
   clientVisit(params, { replace = false } = {}) {
     const current = page.get();
@@ -2253,7 +2261,8 @@ var Router = class {
     };
     const preserveScroll = RequestParams.resolvePreserveOption(params.preserveScroll ?? false, page2);
     const preserveState = RequestParams.resolvePreserveOption(params.preserveState ?? false, page2);
-    page.set(page2, {
+    page.__jodi_isHistoryModal = pageParams.__jodi_isHistoryModal;
+    return page.set(page2, {
       replace,
       preserveScroll,
       preserveState,
@@ -2349,6 +2358,8 @@ var Router = class {
       }),
       onSuccess: options.onSuccess || (() => {
       }),
+      onInvalid: options.onInvalid || (() => {
+      }),
       onError: options.onError || (() => {
       }),
       onPrefetched: options.onPrefetched || (() => {
@@ -2360,7 +2371,7 @@ var Router = class {
   loadDeferredProps(deferred) {
     if (deferred) {
       Object.entries(deferred).forEach(([_, group]) => {
-        this.reload({ only: group });
+        this.reload({ only: group, replace: true });
       });
     }
   }
diff --git a/node_modules/@inertiajs/core/types/router.d.ts b/node_modules/@inertiajs/core/types/router.d.ts
index 92fa7c7..cee9acf 100644
--- a/node_modules/@inertiajs/core/types/router.d.ts
+++ b/node_modules/@inertiajs/core/types/router.d.ts
@@ -29,11 +29,11 @@ export declare class Router {
     clearHistory(): void;
     decryptHistory(): Promise<Page>;
     resolveComponent(component: string): Promise<Component>;
-    replace<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): void;
+    replace<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): Promise<void>;
     replaceProp<TProps = Page['props']>(name: string, value: unknown | ((oldValue: unknown, props: TProps) => unknown), options?: Pick<ClientSideVisitOptions, 'onError' | 'onFinish' | 'onSuccess'>): void;
     appendToProp<TProps = Page['props']>(name: string, value: unknown | unknown[] | ((oldValue: unknown, props: TProps) => unknown | unknown[]), options?: Pick<ClientSideVisitOptions, 'onError' | 'onFinish' | 'onSuccess'>): void;
     prependToProp<TProps = Page['props']>(name: string, value: unknown | unknown[] | ((oldValue: unknown, props: TProps) => unknown | unknown[]), options?: Pick<ClientSideVisitOptions, 'onError' | 'onFinish' | 'onSuccess'>): void;
-    push<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): void;
+    push<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): Promise<void>;
     protected clientVisit<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>, { replace }?: {
         replace?: boolean;
     }): void;
diff --git a/node_modules/@inertiajs/core/types/types.d.ts b/node_modules/@inertiajs/core/types/types.d.ts
index 3fb2f61..1f7232c 100644
--- a/node_modules/@inertiajs/core/types/types.d.ts
+++ b/node_modules/@inertiajs/core/types/types.d.ts
@@ -93,6 +93,7 @@ export interface Page<SharedProps extends PageProps = PageProps> {
     deepMergeProps?: string[];
     matchPropsOn?: string[];
     scrollProps?: Record<keyof PageProps, ScrollProp>;
+    __jodi_isHistoryModal?: boolean;
     /** @internal */
     rememberedState: Record<string, unknown>;
 }
@@ -110,6 +111,7 @@ export interface ClientSideVisitOptions<TProps = Page['props']> {
     preserveState?: VisitOptions['preserveState'];
     errorBag?: string | null;
     viewTransition?: VisitOptions['viewTransition'];
+    __jodi_isHistoryModal?: boolean;
     onError?: (errors: Errors) => void;
     onFinish?: (visit: ClientSideVisitOptions<TProps>) => void;
     onSuccess?: (page: Page) => void;
@@ -157,7 +159,7 @@ export type GlobalEventsMap<T extends RequestPayload = RequestPayload> = {
         details: {
             visit: PendingVisit<T>;
         };
-        result: boolean | void;
+        result: Promise<boolean | void> | boolean | void;
     };
     start: {
         parameters: [PendingVisit<T>];
@@ -263,6 +265,7 @@ export type VisitCallbacks<T extends RequestPayload = RequestPayload> = {
     onCancel: GlobalEventCallback<'cancel', T>;
     onSuccess: GlobalEventCallback<'success', T>;
     onError: GlobalEventCallback<'error', T>;
+    onInvalid: GlobalEventCallback<'invalid', T>;
     onPrefetched: GlobalEventCallback<'prefetched', T>;
     onPrefetching: GlobalEventCallback<'prefetching', T>;
 };
@@ -364,6 +367,7 @@ export interface LinkComponentBaseProps extends Partial<Pick<Visit<RequestPayloa
     prefetch: boolean | LinkPrefetchOption | LinkPrefetchOption[];
     cacheFor: CacheForOption | CacheForOption[];
     cacheTags: string | string[];
+    disableWhileProcessing?: boolean
 }> {
 }
 type PrefetchObject = {
@@ -408,7 +412,7 @@ export type UrlMethodPair = {
     url: string;
     method: Method;
 };
-export type FormComponentOptions = Pick<VisitOptions, 'preserveScroll' | 'preserveState' | 'preserveUrl' | 'replace' | 'only' | 'except' | 'reset'>;
+export type FormComponentOptions = Pick<VisitOptions, 'preserveScroll' | 'preserveState' | 'preserveUrl' | 'replace' | 'only' | 'except' | 'reset' | "viewTransition">;
 export type FormComponentProps = Partial<Pick<Visit, 'headers' | 'queryStringArrayFormat' | 'errorBag' | 'showProgress' | 'invalidateCacheTags'> & Omit<VisitCallbacks, 'onPrefetched' | 'onPrefetching'>> & {
     method?: Method | Uppercase<Method>;
     action?: string | UrlMethodPair;
