diff --git a/node_modules/@inertiajs/svelte/dist/components/Form.svelte b/node_modules/@inertiajs/svelte/dist/components/Form.svelte
index d367d35..ab050e5 100644
--- a/node_modules/@inertiajs/svelte/dist/components/Form.svelte
+++ b/node_modules/@inertiajs/svelte/dist/components/Form.svelte
@@ -38,6 +38,7 @@
     onStart?: FormComponentProps['onStart']
     onProgress?: FormComponentProps['onProgress']
     onFinish?: FormComponentProps['onFinish']
+    onOptimisticRollback?: FormComponentProps['onOptimisticRollback']
     onCancel?: FormComponentProps['onCancel']
     onSuccess?: FormComponentProps['onSuccess']
     onError?: FormComponentProps['onError']
@@ -72,6 +73,7 @@
     onStart = noop,
     onProgress = noop,
     onFinish = noop,
+    onOptimisticRollback = noop,
     onCancel = noop,
     onSuccess = noop,
     onError = noop,
@@ -175,6 +177,7 @@
       onStart,
       onProgress,
       onFinish,
+      onOptimisticRollback,
       onCancel,
       onSuccess: async (...args) => {
         const result = await onSuccess?.(...args)
