diff --git a/node_modules/@inertiajs/core/dist/index.esm.js b/node_modules/@inertiajs/core/dist/index.esm.js
index 7744084..02caba2 100644
--- a/node_modules/@inertiajs/core/dist/index.esm.js
+++ b/node_modules/@inertiajs/core/dist/index.esm.js
@@ -566,12 +566,13 @@ var prefetchedRequests = new PrefetchedRequests();
 // src/scroll.ts
 var Scroll = class {
   static save() {
-    history.saveScrollPositions(
-      Array.from(this.regions()).map((region) => ({
-        top: region.scrollTop,
-        left: region.scrollLeft
-      }))
-    );
+    history.saveScrollPositions(this.getScrollRegions())
+  }
+  static getScrollRegions() {
+    return Array.from(this.regions()).map((region) => ({
+      top: region.scrollTop,
+      left: region.scrollLeft,
+    }))
   }
   static regions() {
     return document.querySelectorAll("[scroll-region]");
@@ -809,9 +810,7 @@ var CurrentPage = class {
       }
       page2.rememberedState ?? (page2.rememberedState = {});
       const isServer2 = typeof window === "undefined";
-      const location = !isServer2 ? window.location : new URL(page2.url);
-      const scrollRegions = !isServer2 && preserveScroll ? history.getScrollRegions() : [];
-      replace = replace || isSameUrlWithoutHash(hrefToUrl(page2.url), location);
+      const scrollRegions = !isServer2 && preserveScroll ? Scroll.getScrollRegions() : [];
       return new Promise((resolve) => {
         replace ? history.replaceState(page2, () => resolve(null)) : history.pushState(page2, () => resolve(null));
       }).then(() => {
@@ -854,13 +853,14 @@ var CurrentPage = class {
     });
   }
   setQuietly(page2, {
-    preserveState = false
+    preserveState = false,
+    viewTransition = false
   } = {}) {
     return this.resolve(page2.component).then((component) => {
       this.page = page2;
       this.cleared = false;
       history.setCurrent(page2);
-      return this.swap({ component, page: page2, preserveState, viewTransition: false });
+      return this.swap({ component, page: page2, preserveState, viewTransition });
     });
   }
   clear() {
@@ -988,7 +988,7 @@ var History = class {
     if (isServer) {
       return;
     }
-    if (this.preserveUrl) {
+    if (this.preserveUrl && (page2.url != page.get().url)) {
       cb && cb();
       return;
     }
@@ -1088,7 +1088,7 @@ var History = class {
     if (isServer) {
       return;
     }
-    if (this.preserveUrl) {
+    if (this.preserveUrl && (page2.url != page.get().url)) {
       cb && cb();
       return;
     }
@@ -1186,6 +1186,9 @@ var EventHandler = class {
       this.internalListeners = this.internalListeners.filter((listener) => listener.listener !== callback);
     };
   }
+  onPopstateFinish() {
+    this.fireInternalEvent("popstateFinish")
+  }
   onMissingHistoryItem() {
     page.clear();
     this.fireInternalEvent("missingHistoryItem");
@@ -1204,23 +1207,37 @@ var EventHandler = class {
       url.hash = window.location.hash;
       history.replaceState({ ...page.get(), url: url.href });
       Scroll.reset();
+      this.onPopstateFinish();
       return;
     }
     if (!history.isValidState(state)) {
-      return this.onMissingHistoryItem();
+      this.onMissingHistoryItem();
+      this.onPopstateFinish()
+      return
     }
     history.decrypt(state.page).then((data) => {
       if (page.get().version !== data.version) {
         this.onMissingHistoryItem();
+        this.onPopstateFinish()
+        return;
+      }
+      if (page.get().__jodi_historyModals?.length) {
+        const { viewTransition } = page.get().__jodi_historyModals.pop();
+        const page2 = { ...page.get(), url: data.url };
+        page.setQuietly(page2, { preserveState: true, viewTransition }).then(() => {
+          this.onPopstateFinish();
+        });
         return;
       }
       router.cancelAll();
       page.setQuietly(data, { preserveState: false }).then(() => {
         Scroll.restore(history.getScrollRegions());
         fireNavigateEvent(page.get());
+        this.onPopstateFinish();
       });
     }).catch(() => {
       this.onMissingHistoryItem();
+      this.onPopstateFinish();
     });
   }
 };
@@ -1314,13 +1331,17 @@ var InitialVisit = class {
     if (typeof window !== "undefined") {
       page.setUrlHash(window.location.hash);
     }
-    page.set(page.get(), { preserveScroll: true, preserveState: true }).then(() => {
+    const page2 = {
+      ...page.get(),
+      __jodi_historyModals: history.initialState?.__jodi_historyModals ?? []
+    };
+    page.set(page2, { preserveScroll: true, preserveState: true, replace: true }).then(() => {
       if (navigationType.isReload()) {
         Scroll.restore(history.getScrollRegions());
       } else {
         Scroll.scrollToAnchor();
       }
-      fireNavigateEvent(page.get());
+      fireNavigateEvent(page2);
     });
   }
 };
@@ -1418,6 +1439,7 @@ var RequestParams = class _RequestParams {
         onCancel: this.wrapCallback(params, "onCancel"),
         onSuccess: this.wrapCallback(params, "onSuccess"),
         onError: this.wrapCallback(params, "onError"),
+        onInvalid: this.wrapCallback(params, "onInvalid"),
         onCancelToken: this.wrapCallback(params, "onCancelToken"),
         onPrefetched: this.wrapCallback(params, "onPrefetched"),
         onPrefetching: this.wrapCallback(params, "onPrefetching")
@@ -1718,7 +1740,7 @@ var Response = class _Response {
       ...this.response,
       data: this.getDataFromResponse(this.response.data)
     };
-    if (fireInvalidEvent(response)) {
+    if (this.requestParams.all().onInvalid(response) !== false && fireInvalidEvent(response)) {
       return config.get("future.useDialogForErrorModal") ? dialog_default.show(response.data) : modal_default.show(response.data);
     }
   }
@@ -1768,6 +1790,7 @@ var Response = class _Response {
     await this.setRememberedState(pageResponse);
     this.requestParams.setPreserveOptions(pageResponse);
     pageResponse.url = history.preserveUrl ? page.get().url : this.pageUrl(pageResponse);
+    pageResponse.__jodi_historyModals = page.get().__jodi_historyModals;
     this.requestParams.all().onBeforeUpdate(pageResponse);
     fireBeforeUpdateEvent(pageResponse);
     return page.set(pageResponse, {
@@ -2057,7 +2080,7 @@ var RequestStream = class {
   }
   send(request) {
     this.requests.push(request);
-    request.send().then(() => {
+    return request.send().then(() => {
       this.requests = this.requests.filter((r) => r !== request);
     });
   }
@@ -2175,13 +2198,13 @@ var Router = class {
       keepAlive: options.keepAlive ?? false
     });
   }
-  visit(href, options = {}) {
+  async visit(href, options = {}) {
     const visit = this.getPendingVisit(href, {
       ...options,
       showProgress: options.showProgress ?? !options.async
     });
     const events = this.getVisitEvents(options);
-    if (events.onBefore(visit) === false || !fireBeforeEvent(visit)) {
+    if (await events.onBefore(visit) === false || !fireBeforeEvent(visit)) {
       return;
     }
     const requestStream = visit.async ? this.asyncRequestStream : this.syncRequestStream;
@@ -2196,11 +2219,10 @@ var Router = class {
     const prefetched = prefetchedRequests.get(requestParams);
     if (prefetched) {
       progress.reveal(prefetched.inFlight);
-      prefetchedRequests.use(prefetched, requestParams);
-    } else {
-      progress.reveal(true);
-      requestStream.send(Request.create(requestParams, page.get()));
+      return prefetchedRequests.use(prefetched, requestParams);
     }
+    progress.reveal(true);
+    return requestStream.send(Request.create(requestParams, page.get()));
   }
   getCached(href, options = {}) {
     return prefetchedRequests.findCached(this.getPrefetchParams(href, options));
@@ -2279,8 +2301,24 @@ var Router = class {
   resolveComponent(component) {
     return page.resolve(component);
   }
+  back() {
+    return new Promise((resolve, reject) => {
+      const unsubscribe = eventHandler.on('popstateFinish', () => {
+        unsubscribe();
+        resolve();
+        clearTimeout(timeout);
+      });
+
+      const timeout = setTimeout(() => {
+        unsubscribe();
+        reject();
+      }, 1000);
+
+      window.history.back();
+    });
+  }
   replace(params) {
-    this.clientVisit(params, { replace: true });
+    return this.clientVisit(params, { replace: true });
   }
   replaceProp(name, value, options) {
     this.replace({
@@ -2320,15 +2358,15 @@ var Router = class {
     );
   }
   push(params) {
-    this.clientVisit(params);
+    return this.clientVisit(params);
   }
   clientVisit(params, { replace = false } = {}) {
-    this.clientVisitQueue.add(() => this.performClientVisit(params, { replace }));
+    return this.clientVisitQueue.add(() => this.performClientVisit(params, { replace }));
   }
   performClientVisit(params, { replace = false } = {}) {
     const current = page.get();
     const props = typeof params.props === "function" ? params.props(current.props) : params.props ?? current.props;
-    const { viewTransition, onError, onFinish, onSuccess, ...pageParams } = params;
+    const { viewTransition, onError, onFinish, onSuccess, __jodi_isHistoryModal, ...pageParams } = params;
     const page2 = {
       ...current,
       ...pageParams,
@@ -2336,6 +2374,9 @@ var Router = class {
     };
     const preserveScroll = RequestParams.resolvePreserveOption(params.preserveScroll ?? false, page2);
     const preserveState = RequestParams.resolvePreserveOption(params.preserveState ?? false, page2);
+    if (__jodi_isHistoryModal && !replace) {
+        (page2.__jodi_historyModals ??= []).push({ viewTransition });
+    }
     return page.set(page2, {
       replace,
       preserveScroll,
@@ -2433,6 +2474,8 @@ var Router = class {
       }),
       onSuccess: options.onSuccess || (() => {
       }),
+      onInvalid: options.onInvalid || (() => {
+      }),
       onError: options.onError || (() => {
       }),
       onPrefetched: options.onPrefetched || (() => {
@@ -2444,7 +2487,7 @@ var Router = class {
   loadDeferredProps(deferred) {
     if (deferred) {
       Object.entries(deferred).forEach(([_, group]) => {
-        this.doReload({ only: group, deferredProps: true });
+        this.doReload({ only: group, deferredProps: true, replace: true, preserveUrl: true });
       });
     }
   }
diff --git a/node_modules/@inertiajs/core/types/router.d.ts b/node_modules/@inertiajs/core/types/router.d.ts
index dcf6009..da6871e 100644
--- a/node_modules/@inertiajs/core/types/router.d.ts
+++ b/node_modules/@inertiajs/core/types/router.d.ts
@@ -6,12 +6,12 @@ export declare class Router {
     protected asyncRequestStream: RequestStream;
     protected clientVisitQueue: Queue<Promise<void>>;
     init<ComponentType = Component>({ initialPage, resolveComponent, swapComponent, }: RouterInitParams<ComponentType>): void;
-    get<T extends RequestPayload = RequestPayload>(url: URL | string | UrlMethodPair, data?: T, options?: VisitHelperOptions<T>): void;
-    post<T extends RequestPayload = RequestPayload>(url: URL | string | UrlMethodPair, data?: T, options?: VisitHelperOptions<T>): void;
-    put<T extends RequestPayload = RequestPayload>(url: URL | string | UrlMethodPair, data?: T, options?: VisitHelperOptions<T>): void;
-    patch<T extends RequestPayload = RequestPayload>(url: URL | string | UrlMethodPair, data?: T, options?: VisitHelperOptions<T>): void;
-    delete<T extends RequestPayload = RequestPayload>(url: URL | string | UrlMethodPair, options?: Omit<VisitOptions<T>, 'method'>): void;
-    reload<T extends RequestPayload = RequestPayload>(options?: ReloadOptions<T>): void;
+    get<T extends RequestPayload = RequestPayload>(url: URL | string | UrlMethodPair, data?: T, options?: VisitHelperOptions<T>): Promise<void>;
+    post<T extends RequestPayload = RequestPayload>(url: URL | string | UrlMethodPair, data?: T, options?: VisitHelperOptions<T>): Promise<void>;
+    put<T extends RequestPayload = RequestPayload>(url: URL | string | UrlMethodPair, data?: T, options?: VisitHelperOptions<T>): Promise<void>;
+    patch<T extends RequestPayload = RequestPayload>(url: URL | string | UrlMethodPair, data?: T, options?: VisitHelperOptions<T>): Promise<void>;
+    delete<T extends RequestPayload = RequestPayload>(url: URL | string | UrlMethodPair, options?: Omit<VisitOptions<T>, 'method'>): Promise<void>;
+    reload<T extends RequestPayload = RequestPayload>(options?: ReloadOptions<T>): Promise<void>;
     protected doReload<T extends RequestPayload = RequestPayload>(options?: ReloadOptions<T> & {
         deferredProps?: boolean;
     }): void;
@@ -24,7 +24,7 @@ export declare class Router {
         stop: VoidFunction;
         start: VoidFunction;
     };
-    visit<T extends RequestPayload = RequestPayload>(href: string | URL | UrlMethodPair, options?: VisitOptions<T>): void;
+    visit<T extends RequestPayload = RequestPayload>(href: string | URL | UrlMethodPair, options?: VisitOptions<T>): Promise<void>;
     getCached(href: string | URL | UrlMethodPair, options?: VisitOptions): InFlightPrefetch | PrefetchedResponse | null;
     flush(href: string | URL | UrlMethodPair, options?: VisitOptions): void;
     flushAll(): void;
@@ -34,11 +34,12 @@ export declare class Router {
     clearHistory(): void;
     decryptHistory(): Promise<Page>;
     resolveComponent(component: string): Promise<Component>;
-    replace<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): void;
+    back(): Promise<void>
+    replace<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): Promise<void>;
     replaceProp<TProps = Page['props']>(name: string, value: unknown | ((oldValue: unknown, props: TProps) => unknown), options?: Pick<ClientSideVisitOptions, 'onError' | 'onFinish' | 'onSuccess'>): void;
     appendToProp<TProps = Page['props']>(name: string, value: unknown | unknown[] | ((oldValue: unknown, props: TProps) => unknown | unknown[]), options?: Pick<ClientSideVisitOptions, 'onError' | 'onFinish' | 'onSuccess'>): void;
     prependToProp<TProps = Page['props']>(name: string, value: unknown | unknown[] | ((oldValue: unknown, props: TProps) => unknown | unknown[]), options?: Pick<ClientSideVisitOptions, 'onError' | 'onFinish' | 'onSuccess'>): void;
-    push<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): void;
+    push<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): Promise<void>;
     protected clientVisit<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>, { replace }?: {
         replace?: boolean;
     }): void;
diff --git a/node_modules/@inertiajs/core/types/types.d.ts b/node_modules/@inertiajs/core/types/types.d.ts
index 3a76ca7..077d175 100644
--- a/node_modules/@inertiajs/core/types/types.d.ts
+++ b/node_modules/@inertiajs/core/types/types.d.ts
@@ -97,6 +97,7 @@ export interface Page<SharedProps extends PageProps = PageProps> {
         prop: keyof PageProps;
         expiresAt?: number | null;
     }>;
+    __jodi_historyModals?: { viewTransition?: boolean }[];
     /** @internal */
     rememberedState: Record<string, unknown>;
 }
@@ -114,6 +115,7 @@ export interface ClientSideVisitOptions<TProps = Page['props']> {
     preserveState?: VisitOptions['preserveState'];
     errorBag?: string | null;
     viewTransition?: VisitOptions['viewTransition'];
+    __jodi_isHistoryModal?: boolean;
     onError?: (errors: Errors) => void;
     onFinish?: (visit: ClientSideVisitOptions<TProps>) => void;
     onSuccess?: (page: Page) => void;
@@ -161,7 +163,7 @@ export type GlobalEventsMap<T extends RequestPayload = RequestPayload> = {
         details: {
             visit: PendingVisit<T>;
         };
-        result: boolean | void;
+        result: Promise<boolean | void> | boolean | void;
     };
     start: {
         parameters: [PendingVisit<T>];
@@ -267,6 +269,7 @@ export type VisitCallbacks<T extends RequestPayload = RequestPayload> = {
     onCancel: GlobalEventCallback<'cancel', T>;
     onSuccess: GlobalEventCallback<'success', T>;
     onError: GlobalEventCallback<'error', T>;
+    onInvalid: GlobalEventCallback<'invalid', T>;
     onPrefetched: GlobalEventCallback<'prefetched', T>;
     onPrefetching: GlobalEventCallback<'prefetching', T>;
 };
@@ -370,6 +373,7 @@ export interface LinkComponentBaseProps extends Partial<Pick<Visit<RequestPayloa
     prefetch: boolean | LinkPrefetchOption | LinkPrefetchOption[];
     cacheFor: CacheForOption | CacheForOption[];
     cacheTags: string | string[];
+    disableWhileProcessing?: boolean
 }> {
 }
 type PrefetchObject = {
@@ -415,7 +419,7 @@ export type UrlMethodPair = {
     url: string;
     method: Method;
 };
-export type FormComponentOptions = Pick<VisitOptions, 'preserveScroll' | 'preserveState' | 'preserveUrl' | 'replace' | 'only' | 'except' | 'reset'>;
+export type FormComponentOptions = Pick<VisitOptions, 'preserveScroll' | 'preserveState' | 'preserveUrl' | 'replace' | 'only' | 'except' | 'reset' | "viewTransition">;
 export type FormComponentProps = Partial<Pick<Visit, 'headers' | 'queryStringArrayFormat' | 'errorBag' | 'showProgress' | 'invalidateCacheTags'> & Omit<VisitCallbacks, 'onPrefetched' | 'onPrefetching'>> & {
     method?: Method | Uppercase<Method>;
     action?: string | UrlMethodPair;
