diff --git a/node_modules/@inertiajs/core/dist/index.esm.js b/node_modules/@inertiajs/core/dist/index.esm.js
index 7744084..fa77db2 100644
--- a/node_modules/@inertiajs/core/dist/index.esm.js
+++ b/node_modules/@inertiajs/core/dist/index.esm.js
@@ -811,7 +811,6 @@ var CurrentPage = class {
       const isServer2 = typeof window === "undefined";
       const location = !isServer2 ? window.location : new URL(page2.url);
       const scrollRegions = !isServer2 && preserveScroll ? history.getScrollRegions() : [];
-      replace = replace || isSameUrlWithoutHash(hrefToUrl(page2.url), location);
       return new Promise((resolve) => {
         replace ? history.replaceState(page2, () => resolve(null)) : history.pushState(page2, () => resolve(null));
       }).then(() => {
@@ -1198,6 +1197,14 @@ var EventHandler = class {
     return () => document.removeEventListener(type, listener);
   }
   handlePopstateEvent(event) {
+    if (page.__jodi_isHistoryModal) {
+        delete page.__jodi_isHistoryModal;
+        const url = hrefToUrl(page.get().url);
+        url.hash = "";
+        const page2 = { ...page.get(), url: url.pathname + url.search };
+        page.setQuietly(page2, { preserveState: true });
+        return;
+    }
     const state = event.state || null;
     if (state === null) {
       const url = hrefToUrl(page.get().url);
@@ -1418,6 +1425,7 @@ var RequestParams = class _RequestParams {
         onCancel: this.wrapCallback(params, "onCancel"),
         onSuccess: this.wrapCallback(params, "onSuccess"),
         onError: this.wrapCallback(params, "onError"),
+        onInvalid: this.wrapCallback(params, "onInvalid"),
         onCancelToken: this.wrapCallback(params, "onCancelToken"),
         onPrefetched: this.wrapCallback(params, "onPrefetched"),
         onPrefetching: this.wrapCallback(params, "onPrefetching")
@@ -1718,7 +1726,7 @@ var Response = class _Response {
       ...this.response,
       data: this.getDataFromResponse(this.response.data)
     };
-    if (fireInvalidEvent(response)) {
+    if (this.requestParams.all().onInvalid(response) !== false && fireInvalidEvent(response)) {
       return config.get("future.useDialogForErrorModal") ? dialog_default.show(response.data) : modal_default.show(response.data);
     }
   }
@@ -2175,13 +2183,13 @@ var Router = class {
       keepAlive: options.keepAlive ?? false
     });
   }
-  visit(href, options = {}) {
+  async visit(href, options = {}) {
     const visit = this.getPendingVisit(href, {
       ...options,
       showProgress: options.showProgress ?? !options.async
     });
     const events = this.getVisitEvents(options);
-    if (events.onBefore(visit) === false || !fireBeforeEvent(visit)) {
+    if (await events.onBefore(visit) === false || !fireBeforeEvent(visit)) {
       return;
     }
     const requestStream = visit.async ? this.asyncRequestStream : this.syncRequestStream;
@@ -2280,7 +2288,7 @@ var Router = class {
     return page.resolve(component);
   }
   replace(params) {
-    this.clientVisit(params, { replace: true });
+    return this.clientVisit(params, { replace: true });
   }
   replaceProp(name, value, options) {
     this.replace({
@@ -2320,7 +2328,7 @@ var Router = class {
     );
   }
   push(params) {
-    this.clientVisit(params);
+    return this.clientVisit(params);
   }
   clientVisit(params, { replace = false } = {}) {
     this.clientVisitQueue.add(() => this.performClientVisit(params, { replace }));
@@ -2336,6 +2344,7 @@ var Router = class {
     };
     const preserveScroll = RequestParams.resolvePreserveOption(params.preserveScroll ?? false, page2);
     const preserveState = RequestParams.resolvePreserveOption(params.preserveState ?? false, page2);
+    page.__jodi_isHistoryModal = pageParams.__jodi_isHistoryModal;
     return page.set(page2, {
       replace,
       preserveScroll,
@@ -2433,6 +2442,8 @@ var Router = class {
       }),
       onSuccess: options.onSuccess || (() => {
       }),
+      onInvalid: options.onInvalid || (() => {
+      }),
       onError: options.onError || (() => {
       }),
       onPrefetched: options.onPrefetched || (() => {
@@ -2444,7 +2455,7 @@ var Router = class {
   loadDeferredProps(deferred) {
     if (deferred) {
       Object.entries(deferred).forEach(([_, group]) => {
-        this.doReload({ only: group, deferredProps: true });
+        this.doReload({ only: group, deferredProps: true, replace: true });
       });
     }
   }
diff --git a/node_modules/@inertiajs/core/types/router.d.ts b/node_modules/@inertiajs/core/types/router.d.ts
index dcf6009..cb3bd96 100644
--- a/node_modules/@inertiajs/core/types/router.d.ts
+++ b/node_modules/@inertiajs/core/types/router.d.ts
@@ -34,11 +34,11 @@ export declare class Router {
     clearHistory(): void;
     decryptHistory(): Promise<Page>;
     resolveComponent(component: string): Promise<Component>;
-    replace<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): void;
+    replace<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): Promise<void>;
     replaceProp<TProps = Page['props']>(name: string, value: unknown | ((oldValue: unknown, props: TProps) => unknown), options?: Pick<ClientSideVisitOptions, 'onError' | 'onFinish' | 'onSuccess'>): void;
     appendToProp<TProps = Page['props']>(name: string, value: unknown | unknown[] | ((oldValue: unknown, props: TProps) => unknown | unknown[]), options?: Pick<ClientSideVisitOptions, 'onError' | 'onFinish' | 'onSuccess'>): void;
     prependToProp<TProps = Page['props']>(name: string, value: unknown | unknown[] | ((oldValue: unknown, props: TProps) => unknown | unknown[]), options?: Pick<ClientSideVisitOptions, 'onError' | 'onFinish' | 'onSuccess'>): void;
-    push<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): void;
+    push<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>): Promise<void>;
     protected clientVisit<TProps = Page['props']>(params: ClientSideVisitOptions<TProps>, { replace }?: {
         replace?: boolean;
     }): void;
diff --git a/node_modules/@inertiajs/core/types/types.d.ts b/node_modules/@inertiajs/core/types/types.d.ts
index 3a76ca7..1aa779f 100644
--- a/node_modules/@inertiajs/core/types/types.d.ts
+++ b/node_modules/@inertiajs/core/types/types.d.ts
@@ -97,6 +97,7 @@ export interface Page<SharedProps extends PageProps = PageProps> {
         prop: keyof PageProps;
         expiresAt?: number | null;
     }>;
+    __jodi_isHistoryModal?: boolean;
     /** @internal */
     rememberedState: Record<string, unknown>;
 }
@@ -114,6 +115,7 @@ export interface ClientSideVisitOptions<TProps = Page['props']> {
     preserveState?: VisitOptions['preserveState'];
     errorBag?: string | null;
     viewTransition?: VisitOptions['viewTransition'];
+    __jodi_isHistoryModal?: boolean;
     onError?: (errors: Errors) => void;
     onFinish?: (visit: ClientSideVisitOptions<TProps>) => void;
     onSuccess?: (page: Page) => void;
@@ -161,7 +163,7 @@ export type GlobalEventsMap<T extends RequestPayload = RequestPayload> = {
         details: {
             visit: PendingVisit<T>;
         };
-        result: boolean | void;
+        result: Promise<boolean | void> | boolean | void;
     };
     start: {
         parameters: [PendingVisit<T>];
@@ -267,6 +269,7 @@ export type VisitCallbacks<T extends RequestPayload = RequestPayload> = {
     onCancel: GlobalEventCallback<'cancel', T>;
     onSuccess: GlobalEventCallback<'success', T>;
     onError: GlobalEventCallback<'error', T>;
+    onInvalid: GlobalEventCallback<'invalid', T>;
     onPrefetched: GlobalEventCallback<'prefetched', T>;
     onPrefetching: GlobalEventCallback<'prefetching', T>;
 };
@@ -370,6 +373,7 @@ export interface LinkComponentBaseProps extends Partial<Pick<Visit<RequestPayloa
     prefetch: boolean | LinkPrefetchOption | LinkPrefetchOption[];
     cacheFor: CacheForOption | CacheForOption[];
     cacheTags: string | string[];
+    disableWhileProcessing?: boolean
 }> {
 }
 type PrefetchObject = {
@@ -415,7 +419,7 @@ export type UrlMethodPair = {
     url: string;
     method: Method;
 };
-export type FormComponentOptions = Pick<VisitOptions, 'preserveScroll' | 'preserveState' | 'preserveUrl' | 'replace' | 'only' | 'except' | 'reset'>;
+export type FormComponentOptions = Pick<VisitOptions, 'preserveScroll' | 'preserveState' | 'preserveUrl' | 'replace' | 'only' | 'except' | 'reset' | "viewTransition">;
 export type FormComponentProps = Partial<Pick<Visit, 'headers' | 'queryStringArrayFormat' | 'errorBag' | 'showProgress' | 'invalidateCacheTags'> & Omit<VisitCallbacks, 'onPrefetched' | 'onPrefetching'>> & {
     method?: Method | Uppercase<Method>;
     action?: string | UrlMethodPair;
