diff --git a/node_modules/@zag-js/drawer/dist/drawer.machine.mjs b/node_modules/@zag-js/drawer/dist/drawer.machine.mjs
index 4353b22..8353dc8 100644
--- a/node_modules/@zag-js/drawer/dist/drawer.machine.mjs
+++ b/node_modules/@zag-js/drawer/dist/drawer.machine.mjs
@@ -267,7 +267,7 @@ var machine = createMachine({
         CLOSE: [
           {
             guard: "isOpenControlled",
-            actions: ["clearSwipeOpenAnimation", "invokeOnClose"]
+            actions: ["invokeOnClose"]
           },
           {
             target: "closing",
